src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CachingFilePathComparer.cs

```diff
@@ -109,14 +109,24 @@ private unsafe int GetNonRandomizedHashCodeOrdinalIgnoreCase(string obj)
                         goto NotAscii;
                     }
 
-                    hash2 = (RuntimeBitOperations.RotateLeft(hash2, 5) + hash2) ^ (p0 | NormalizeToLowercase);
+                    hash2 = (RotateLeft(hash2, 5) + hash2) ^ (p0 | NormalizeToLowercase);
                 }
             }
 
             return (int)(hash1 + (hash2 * 1566083941));
 
 NotAscii:
             return s_comparer.GetHashCode(obj);
+
```

Comment:
I'd add BitConversions to MS.CA.Contracts project. Similar to https://github.com/dotnet/roslyn/blob/main/src/Dependencies/Contracts/ValueTaskExtensions.cs.

That way we can call the dotnet version directly and net472 polyfill will be applied as needed.